Parallel pattern matcher

vsbogd opened the pull request:
Tuning to measure parallel pattern matcher performance.

ngeiswei commented:
@vsbogd Could you as well update https://github.com/opencog/benchmark/blob/master/atomspace/atomspace/README.md to reflect the new path?

ngeiswei commented:
That looks good to me, just update the README.md of the atomspace benchmark since you've touched that part.

linas commented:
This pull req is OK by me, I guess.

Please be aware that the benchmark code here is incredibly ugly, filled with hacks, and is in need of some large rewrite/redesign.  Some of the things it measures no longer exist (e.g. the distinction between the table and the atomspace is gone; they are essentially the same, now)

Also, I am not convinced that the way it counts time is correct.  It's filled it with a lot of beginner errors; the person who created this had no experience in benchmarking.

vsbogd commented:
Fixed documentation and added an issue https://github.com/opencog/benchmark/issues/6 to use some standard C++ benchmarking library to simplify adding benchmarks.
